Public/Sdk/Public/Protocols/Grpc/protoc.dsc

```diff
@@ -24,44 +24,80 @@ export const tool: Transformer.ToolDefinition = {
+        let dependencies = [
+            ...protoFiles,
+            ...(args.dependencies || [])
+        ];
+
+        let results = (<File[]>protoFiles).map(p => generate({
+            proto: p,
+            enableGrpc: args.enableGrpc,
+            dependencies: dependencies,
+        }));
+
+        return {
+            sources: results.mapMany(result => result.sources)
+        };
+    }
+
+    const protoFile = <File>args.proto;
+
+    const enableGrpc = args.enableGrpc !== false;
+
     const outputDirectory = Context.getNewOutputDirectory("protobuf");
     const arguments : Argument[] = [
-        Cmd.option("--proto_path ", Artifact.none(args.proto.parent)),
+        Cmd.option("--proto_path ", Artifact.none(protoFile.parent)),
         Cmd.option("--csharp_out ", Artifact.none(outputDirectory)),
-        Cmd.files([args.proto]),
-        Cmd.option("--grpc_out ", Artifact.none(outputDirectory)),
-        Cmd.option("--plugin=protoc-gen-grpc=", Artifact.input(pkgContents.getFile(
-            Context.getCurrentHost().os === "win"
-                ? r`tools/windows_x64/grpc_csharp_plugin.exe`
-                : r`tools/macosx_x64/grpc_csharp_plugin`)
+        Cmd.files([protoFile]),
+        ...addIf(enableGrpc,
+            Cmd.option("--grpc_out ", Artifact.none(outputDirectory)),
+            Cmd.option("--plugin=protoc-gen-grpc=", Artifact.input(pkgContents.getFile(
+                Context.getCurrentHost().os === "win"
+                    ? r`tools/windows_x64/grpc_csharp_plugin.exe`
+                    : r`tools/macosx_x64/grpc_csharp_plugin`)
+                )
             )
         ),
+        Cmd.options("--proto_path=", Artifact.inputs(args.includes)),
     ];
 
-    let mainCsFile = p`${outputDirectory}/${args.proto.nameWithoutExtension + ".cs"}`;
-    let grpcCsFile = p`${outputDirectory}/${args.proto.nameWithoutExtension + "Grpc.cs"}`;
+    let mainCsFile = p`${outputDirectory}/${protoFile.nameWithoutExtension + ".cs"}`;
+    let grpcCsFile = p`${outputDirectory}/${protoFile.nameWithoutExtension + "Grpc.cs"}`;
 
     let result = Transformer.execute({
         tool: args.tool || tool,
         arguments: arguments,
         workingDirectory: outputDirectory,
         outputs: [
             mainCsFile,
-            grpcCsFile,
+            ...addIf(enableGrpc,
+                grpcCsFile
+            ),
+        ],
+        dependencies: [
+            ...(args.dependencies || []),
+            ...addIfLazy(Context.getCurrentHost().os === "macOS", () => MacOS.filesAndSymlinkInputDeps)
         ]
     });
 
     return {
         sources: [
             result.getOutputFile(mainCsFile),
-            result.getOutputFile(grpcCsFile),
+            ...addIf(enableGrpc,
+                result.getOutputFile(grpcCsFile)
+            ),
         ],
     };
 }
 
 @@public
 export interface Arguments extends Transformer.RunnerArguments{
-    proto: File,
+    proto: File | File[],

```

Comment:
Is this File | File[] actually needed? Just makes the logic more complicated. Can't we live with File[] only?

Reply:
sure